lighthouse-logger/index.js

```diff
@@ -6,12 +6,54 @@
 'use strict';
 
 const debug = require('debug');
+// @ts-ignore
 const marky = require('marky');
-
 const EventEmitter = require('events').EventEmitter;
+
+/**
+ * @template {Function} F
+ * @typedef {import('./type-helpers').ArgumentTypes<F>} ArgumentTypes
+ */
+
+/**
+ * @template T
+ * @typedef {import('./type-helpers').IsFunction<T>} IsFunction
+ */
+
+/**
+ * @template Class
+ * @template {*[]} Args
+ * @typedef {{
+ *  msg: string | ((this: Class, ...args: Args) => string),
+ *  id?: string | ((this: Class, ...args: Args) => string),
```

Comment:
what if we drop the dynamic messages and only support strings here (at least for now)? It would make this file simpler to read, simpler for type checking, and reduces the non-local reasoning needed where the `timeDecorateClass` calls are located.

AFAICT, the only places we need dynamically created strings are for
- running beforePass/pass/afterPass for a gatherer, and
- running an audit
- computing a computed artifact.

Due to the structure of `gatherRunner` there's no good place to decorate the gatherer functions, so #6515 only included dynamic messages for last two. We could do those with manual `log.time()`/`log.timeEnd()` calls (only two pairs of them needed) and be able to get rid of a lot of machinery in here.

And then at a later date we can always add support if we find we want it.

Reply:
It'll be most immediately needed by #6410, since each call to `sendCommand` has a timing with an id derived from the input. As soon as this lands I'll be pushing that issue forward again. With that in mind, should this stay?

Reply:
hmm, yeah, that's useful enough that I guess we should probably keep it